src/fsharp/FSharp.Core/string.fs

```diff
@@ -12,6 +12,11 @@ namespace Microsoft.FSharp.Core
     [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
     [<RequireQualifiedAccess>]
     module String =
+        [<Literal>]
+        /// LOH threshold is calculated from FSharp.Compiler.AbstractIL.Internal.Library.LOH_SIZE_THRESHOLD_BYTES,
+        /// and is equal to 80_000 / sizeof<char>
+        let LOH_CHAR_THRESHOLD = 40_000
```

Comment:
Is hardcoding LOH size a good idea? Because it can configured with COMPlus_GCLOHThreshold

Reply:
@liminiens I'm not aware of that configuration option, the MS documentation I've seen talks of a hardcoded limit of 85,000 bytes exactly.

I followed the same course as done elsewhere for `ResizeArray`. If you know of a method that can get this value, we can certainly update to use that.

Also note that this is currently not common: most library functions are not written to try to avoid the LOH. But maybe that should change.

Reply:
Another note is that operating on strings is very common, including large ones. They tend to be places where LOH allocations occur without you noticing. We can also comb through the entire codebase to trim up this sort of stuff if need be.